cram/cram_codecs.c

```diff
@@ -1400,18 +1400,21 @@ cram_codec *cram_xpack_decode_init(cram_block_compression_hdr *hdr,
         c->decode = cram_xpack_decode_char;
     else {
         fprintf(stderr, "BYTE_ARRAYs not supported by this codec\n");
-        return NULL;
+        goto malformed;
     }
     c->free = cram_xpack_decode_free;
     c->size = cram_xpack_decode_size;
     c->get_block = cram_xpack_get_block;
 
     c->u.xpack.nbits = vv->varint_get32(&cp, endp, NULL);
     c->u.xpack.nval  = vv->varint_get32(&cp, endp, NULL);
+    if (c->u.xpack.nval >= 256)
```

Comment:
40afaaa already added a similar check to this, but used `>` instead of `>=`.  Looking at the loop a few lines below, a value of 256 ought to be OK, so would it be best to resolve the conflict in favour of the version in 40afaaa?

Reply:
I thought it felt familier!  However at the time I mistakenly believed my other PR had been merged - losing track as so many outstanding things being juggled.

Agreed a value of 256 would be valid, so I'll fix it (or you can if you wish?).  It shouldn't happen though as there's no point to any value > 16 really as you're just packing 1 byte to 1 byte.  It's valid, just pointless.

Reply:
I have a fixed version now, so I'll push it up.